Add importer done page

yscik opened the pull request:
### Changes proposed in this Pull Request

* Add components to render the Done page and Import log
* Add done step state to store with results and import log
* Add restart importer action that resets state
* Adjust progress polling to dispatch an action instead of step data invalidation and resolver - so it also updates the done step
* CSS: Make setup wizard grey links explicit so they are not the default

### Testing instructions

* Import some content (questions) without errors
* Done page should show success, and the right number of imported questions
* `Import More Content` should go back to the first step and allow a new import
* Import a questions csv with errors (for example, invalid media attachments)
* Done page should show a warning, the right number for successful and failed imports
* Import log should show detailed description for the affected entries

<!--
Helpful tips for screenshots:
https://en.support.wordpress.com/make-a-screenshot/
-->
### Screenshot / Video

![image](https://user-images.githubusercontent.com/176949/86045471-f68d7280-ba4b-11ea-9bba-9c4d3039877b.png)

![image](https://user-images.githubusercontent.com/176949/86046424-7f58de00-ba4d-11ea-9a85-ffd961c1eacb.png)

yscik commented:
@donnapep  Should the big green success icon be something else if there was only failure? And it could also be a bit confusing if the import was only partially successful.

yscik commented:
Notes:
* The import log is a bit narrow with current message lengths, might be worth exploring making this page full width.
* Currently no way to get back to the import done screen when clicking on `X questions` links - user might expect these just show a list of the imported entries in-place, instead of taking them away.

donnapep commented:
> * Currently no way to get back to the import done screen when clicking on `X questions` links - user might expect these just show a list of the imported entries in-place, instead of taking them away.

Yeah, good call. Perhaps for now we could just remove the links?

yscik commented:
* Removed links
* Also updated to not show summaries where it's empty (0 lessons), and not show the success group if there weren't any. 
* Added tests

jom commented on assets/data-port/import/data/actions.js:
```diff
@@ -315,3 +350,26 @@ export const setStepData = ( step, data ) => ( {
 	data,
 } );
+
+export function* fetchImportLog( jobId ) {
```
Missing docblock.

jom commented on assets/data-port/import/data/actions.js:
```diff
@@ -315,3 +350,26 @@ export const setStepData = ( step, data ) => ( {
 	data,
 } );
+
+export function* fetchImportLog( jobId ) {
```
Missing docblock.

yscik replied:
Fixed this, somewhere  😳

jom commented on assets/data-port/import/data/reducer.js:
```diff
@@ -168,12 +179,15 @@ export default ( state = DEFAULT_STATE, action ) => {
 		case SET_STEP_DATA:
```
Without the resolver, I think this is only used in `fetchImportLog`. I wonder if we should just rename the action, constant, reducer use to `SUCCCESS_FETCH_IMPORT_LOG`?

yscik replied:
Replaced in 72528b2

jom commented on assets/data-port/import/done/done-page.js:
```diff
@@ -0,0 +1,85 @@
+import { __ } from '@wordpress/i18n';
+import { Section } from '@woocommerce/components';
+import { useEffect } from '@wordpress/element';
+import { formatString } from '../../../setup-wizard/helpers/format-string';
```
We might want to consider moving this to a shared namespace

yscik replied:
Good catch, 5348b32

jom commented on assets/data-port/import/done/index.js:
```diff
@@ -1,18 +1,24 @@
+
+		return {
+			fetchImportLog: () =>
+				fetchImportLog( select( 'sensei/import' ).getJobId() ),
```
Nice! I wonder if we could do something like this with `_.partial` for the other places we are passing down the job ID.

yscik replied:
Yeah that looks right. 375b739

jom replied:
I don't think `_.partial` is needed unless there are other arguments, but doesn't harm anything either I suppose.

yscik replied:
One extra benefit here is the `jobId` is selected at render time, not when calling the function. Well, it feels like that's a benefit :)
Also easy to extend if something like paging parameters come in.

yscik commented on assets/data-port/import/data/actions.js:
```diff
@@ -297,21 +288,34 @@ export const errorFileUpload = ( level, error ) => ( {
+ *
+ * @param {string} jobId The job id.
  */
+export const fetchImportLog = composeFetchAction(
+	START_FETCH_IMPORT_LOG,
+	function*( jobId ) {
+		const data = yield fetchFromAPI( {
+			path: buildJobEndpointUrl( jobId, [ 'logs' ] ),
+		} );
+
+		return data;
+	},
+	SUCCESS_FETCH_IMPORT_LOG,
+	ERROR_FETCH_IMPORT_LOG
+);
```
I added a small helper for this START, fetch, SUCCESS, ERROR pattern. It removes the individual actions creators that are mostly just noise, but still keeps the action types explicit.

jom replied:
I like it.

jom commented on assets/data-port/import/data/actions.js:
```diff
@@ -36,65 +41,51 @@ export const fetchFromAPI = ( request ) => ( {
 		const data = yield fetchFromAPI( {
-			path: buildJobEndpointUrl( API_SPECIAL_ACTIVE_JOB_ID ),
+			//path: buildJobEndpointUrl( API_SPECIAL_ACTIVE_JOB_ID ),
+			path: buildJobEndpointUrl( '3efc3b695a2a85c2ac635923cc604d4c' ),
```
🔢 🔢 🔢 🔢

yscik replied:
Wow, how did THAT get in there? 😱 
Decreased number of hashes in 7e23991

jom replied:
😆

yscik commented:
Yeah, I didn't want to complicate that, since the logs are fetched when opening the page and are hidden by default, it should be already loaded by the time users open the section in 99% of cases.
